servers/cromwell/jobs/controllers/jobs_controller.py

```diff
@@ -231,12 +231,11 @@ def format_scattered_task(task_name, task_metadata):
     filtered_shards = []
     current_shard = ''
     min_start = _parse_datetime(task_metadata[0].get('start'))
+    max_end = _parse_datetime(task_metadata[-1].get('end'))
 
     # go through calls in reverse to grab the latest attempt if there are multiple
     for shard in task_metadata[::-1]:
```

Comment:
[ToL] This is well beyond the scope of this PR but whenever I see a construct like
```
biggest_so_far = 0
for thing in stuff:
  if some_function(thing) > biggest_so_far:
    biggest_so_far = some_function(thing)
```
I wonder if we could eliminate state by doing something like
```
biggest = max(map(some_function, stuff))
```